projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/case-flag/components/select-flag-type/select-flag-type.component.ts

```diff
@@ -91,12 +93,23 @@ export class SelectFlagTypeComponent implements OnInit, OnDestroy {
           error: error => this.onRefdataError(error)
         });
     }
+    this.caseValidationErrorsSubscription$ = this.caseEditDataService.caseFormValidationErrors$.subscribe({
+      next: (validationErrors) => {
+        if (validationErrors.length) {
+          this.errorMessages = [];
+          this.flagTypeNotSelectedErrorMessage = '';
+          this.flagTypeErrorMessage = '';
+          this.errorMessages = [];
+        }
+      }
+    });
   }
 
   public ngOnDestroy(): void {
     if (this.flagRefdata$) {
       this.flagRefdata$.unsubscribe();
     }
+    this.caseValidationErrorsSubscription$.unsubscribe();
```

Comment:
@Kartheek-S - as before, probably should check the subscription exists before calling `unsubscribe()`